metabase/metabase
# Consistently handle infinite and empty values in scalar visualizations

Closes https://github.com/metabase/metabase/issues/27395

### Description

- organizes Gauge and Progress visualizations into their own directories
- adds easily-testable getValue functions for these visualizations

the visualizations should show the following values for each case


  | number | gauge | progress
-- | -- | -- | --
0 | 0 | 0 | 0
null | null | 0 | 0
NaN | NaN | 0 | 0
infinity | infinity | Infinity (full gauge) | infinity (full bar)

### How to verify

- create a few visualizations using the sample query in the issue above
- see that the visualizations produce the expected output in each case

### Demo

![Screen Shot 2023-04-03 at 5 02 01 PM](https://user-images.githubusercontent.com/30528226/229856047-77a75c8d-d499-4bb4-b2d9-e8473803a21a.png)
![Screen Shot 2023-04-03 at 5 01 54 PM](https://user-images.githubusercontent.com/30528226/229856049-2dfa59f8-9ee0-4048-8d4f-cbf00c5b6b2a.png)
![Screen Shot 2023-04-03 at 5 01 46 PM](https://user-images.githubusercontent.com/30528226/229856050-44b97f63-98eb-410a-9df7-ab9af68d4e2b.png)
![Screen Shot 2023-04-03 at 5 01 39 PM](https://user-images.githubusercontent.com/30528226/229856052-2053cd4f-5d69-4504-8b34-ea75b5f823ff.png)
![Screen Shot 2023-04-03 at 5 01 32 PM](https://user-images.githubusercontent.com/30528226/229856055-b34f284e-f004-4c5e-ad0c-550547f465e0.png)
![Screen Shot 2023-04-03 at 5 01 25 PM](https://user-images.githubusercontent.com/30528226/229856057-6680b1e3-07a4-4a64-b0ef-17bceba593f7.png)


### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29805)
